### DISCUSSION TITLE
staticcheck on Ubuntu

### DISCUSSION THREAD

**Discussion by ousloob** [dominikh/go-tools#1427] [Q&A] [answered]
I've installed staticcheck by doing go install command
The package is well installed inside the $GOPATH/bin

But when I do the command `staticcheck ./...` I have :
`staticcheck: command not found`
Am I missing something?

Running on Ubuntu

**Comment by atc0005** [comment, marked as answer]
What happens if you run:

- `$GOPATH/bin/staticcheck --version`
- `$(go env GOPATH)/bin/staticcheck --version`
- `which staticcheck`
  - this is expected to fail if `$GOPATH/bin` is not in your path

It sounds like `$GOPATH/bin` is not in your path.

A line like this at the bottom of your `~/.profile` file should resolve that:

```shell
export PATH=$PATH:$(go env GOPATH)/bin
```

Then run:

`source ~/.profile` within your terminal (or log out and back in again) to set the updated `PATH` value.

These commands should now work:

- `which staticcheck`
- `staticcheck --version`

**Comment by ousloob** [comment]
- `$GOPATH/bin/staticcheck --version` => Nothing
- `$(go env GOPATH)/bin/staticcheck --version` => staticcheck 2023.1.3 (v0.4.3)
- `which staticcheck` => Nothing

Your solution worked ! Thank you !
In the go installation instruction they say that we should add:
`export PATH=$PATH:/usr/local/go/bin`

They never talked about the `$GOPATH/bin`

So if I need to execute some go code inside the bin directory I need to add it on my path right?

So maybe I should do this then:
```Shell
export PATH=$PATH:/usr/local/go/bin:$(go env GOPATH)/bin`
```

**Comment by atc0005** [comment]
> In the go installation instruction they say that we should add: `export PATH=$PATH:/usr/local/go/bin`
> 
> They never talked about the `$GOPATH/bin`

I thought the official docs covered adding `$GOPATH/bin`, but when I looked earlier I didn't spot it.

I did find this in my bookmarks:

- https://www.digitalocean.com/community/tutorials/how-to-install-go-and-set-up-a-local-programming-environment-on-ubuntu-18-04

They cover updating `$PATH` via editing `$HOME/.profile` and I'm pretty sure I used that guide years ago.

> So if I need to execute some go code inside the bin directory I need to add it on my path right?

Or fully qualify the command when you run it, which would get annoying. :)

> So maybe I should do this then:
> 
> ```shell
> export PATH=$PATH::/usr/local/go/bin:$(go env GOPATH)/bin`
> ```

That looks like it would work to me.

**Reply by ousloob** [reply]
I think they coverd adding `$GOPATH/bin` before the go module thing.

Thank you for sharing :)
